openmdao/solvers/petsc_ksp.py

```diff
@@ -187,6 +193,11 @@ def solve(self, rhs_mat, system, mode):
 
         for voi, rhs in iteritems(rhs_mat):
 
+            # We have to resize our KSP matrix for each voi if we are using
+            # relevance reduction.
+            if voi is not None:
+                self.setup(system, voi)
```

Comment:
I think it would be better to call setup once for each voi and cache the results in a dict keyed on voi.  Doing setup for each voi for every solve is going to be expensive.